rrule/src/iter/iterinfo.rs

```diff
@@ -125,8 +125,13 @@ impl<'a> IterInfo<'a> {
     pub fn weekday_set(&self, year: i32, month: u32, day: u32) -> Vec<usize> {
         let set_len = usize::from(self.year_len() + 7);
 
-        let mut date_ordinal = usize::try_from(chrono::Utc.ymd(year, month, day).ordinal0())
-            .expect("target arch should have at least 32 bits");
+        let mut date_ordinal = usize::try_from(
+            chrono::Utc
+                .with_ymd_and_hms(year, month, day, 0, 0, 0)
+                .unwrap()
```

Comment:
Same as above.
But anyway, we need to return this result and handle it.
This one _can_ fail.

Reply:
Why I didn't handle them now... because they cannot be worse than before.
Before they could also fail in panic!

Reply:
Check the `unwrap` in the deprecated `ymd` function.
![image](https://user-images.githubusercontent.com/45714/226856818-2a1eb8ac-3f47-43ad-b89a-ecfcc6e457b8.png)

Reply:
I agree with you. Do you mind adding a TODO comment here about the possible panic, so that the chance of this being fixed in the future will increase a tiny bit? :)

Reply:
I don't like TODOs in the code. It's so easy to forget them. I'll create a ticket on GitHub

Reply:
Here is our deprecated impl:
![Screenshot from 2023-03-25 08-55-02](https://user-images.githubusercontent.com/45714/227705095-202b5d0d-d3ca-4cbf-856e-957c6266fc44.png)

Now when I go to `chrono`'s `Local` impl, you see they use a deprecated method also (My IDE didn't mark it as deprecared, but it is the one!)
And the whole trait method is also marked as deprecated.
![Screenshot from 2023-03-25 08-55-42](https://user-images.githubusercontent.com/45714/227705129-f7f9e006-b798-444c-b0e2-323098c2b7a7.png)


AFAICS, this `impl TimeZone for` we have, is kinda combination of its impl for chrono's `Local` and chorono-tz's `Tz`, which you can find them [here](https://github.com/chronotope/chrono/blob/main/src/offset/local/mod.rs#L113) and [here](https://github.com/chronotope/chrono-tz/blob/main/chrono-tz/src/timezone_impl.rs#L343).

In the second one, you can see how they solved it, but I think generally spending time on something deprecated which will be removed soon from the main `trait` is useless.

Reply:
here is the issue for these: https://github.com/fmeringdal/rust-rrule/issues/90